src/views/Dashboard/Customer/Home/PaymentHistory/index.vue

```diff
@@ -73,13 +73,23 @@ export default {
 
   computed: {
     ...mapState({
-      web3: (state) => state.metamask.web3
+      web3: (state) => state.metamask.web3,
+      lastEventData: (state) => state.substrate.lastEventData
     })
   },
 
   watch: {
     searchQuery(newVal, oldVal) {
       if (newVal === "" && oldVal) this.metamaskDispatchAction(this.onSearchInput)
+    },
+    
+    lastEventData(event) {
+      if (event !== null) {
+        const methodToRefetch = ["OrderCreated", "OrderCancelled"]
+        if (methodToRefetch.includes(event.method)) {
+          fetchPaymentHistories()
+        }
+      }
```

Comment:
make it simpler by using return when the value is null, try not to write much logic in on the `if` block condition. Let's make if condition simpler like this one

```js
if (!event) return // or you want to make it like this if (event === null) whatever
const methodToRefetch = ["OrderCreated", "OrderCancelled"]
if (methodToRefetch.includes(event.method)) fetchPaymentHistories()
```